pkg/allocation/converters/converter.go

```diff
@@ -206,28 +210,14 @@ func convertAllocationPortsV1Alpha1ToAgonesPortsV1(in []*pb.AllocationResponse_G
 }
 
 // convertStateV1ToAllocationStateV1Alpha1 converts GameServerAllocationState V1 (GSA) to AllocationResponse_GameServerAllocationState
-func convertStateV1ToAllocationStateV1Alpha1(in allocationv1.GameServerAllocationState) pb.AllocationResponse_GameServerAllocationState {
+func convertStateV1ToError(in allocationv1.GameServerAllocationState) error {
 	switch in {
 	case allocationv1.GameServerAllocationAllocated:
-		return pb.AllocationResponse_Allocated
+		return nil
 	case allocationv1.GameServerAllocationUnAllocated:
-		return pb.AllocationResponse_UnAllocated
+		return status.Error(codes.ResourceExhausted, "there isn't any available Game Server to allocate")
```

Comment:
```suggestion
		return status.Error(codes.ResourceExhausted, "there is no available Game Server to allocate")
```
or could be 
`"there aren't any available Game Servers to allocate"`

Reply:
Done.